internal/query/auth_request.go

```diff
@@ -0,0 +1,70 @@
+package query
+
+import (
+	"context"
+	"database/sql"
+	_ "embed"
+	errs "errors"
+	"fmt"
+	"time"
+
+	"github.com/zitadel/zitadel/internal/api/call"
+	"github.com/zitadel/zitadel/internal/database"
+	"github.com/zitadel/zitadel/internal/domain"
+	"github.com/zitadel/zitadel/internal/errors"
+	"github.com/zitadel/zitadel/internal/query/projection"
+	"github.com/zitadel/zitadel/internal/telemetry/tracing"
+)
+
+type AuthRequest struct {
+	ID           string
+	CreationDate time.Time
+	ClientID     string
+	Scope        []string
+	RedirectURI  string
+	Prompt       []domain.Prompt
+	UiLocales    []string
+	LoginHint    *string
+	MaxAge       *time.Duration
+	HintUserID   *string
+}
+
+//go:embed embed/auth_request_by_id.sql
+var authRequestByIDQuery string
+
+func (q *Queries) authRequestByIDQuery(ctx context.Context) string {
+	return fmt.Sprintf(authRequestByIDQuery, q.client.Timetravel(call.Took(ctx)))
+}
+
+func (q *Queries) AuthRequestByID(ctx context.Context, shouldTriggerBulk bool, id string) (_ *AuthRequest, err error) {
+	ctx, span := tracing.NewSpan(ctx)
+	defer func() { span.EndWithError(err) }()
+
+	if shouldTriggerBulk {
+		projection.AuthRequestProjection.Trigger(ctx)
+	}
+
+	var (
+		scope   database.StringArray
+		prompt  database.EnumArray[domain.Prompt]
+		locales database.StringArray
+	)
+
+	dst := new(AuthRequest)
+	err = q.client.DB.QueryRowContext(ctx, q.authRequestByIDQuery(ctx), id).Scan(
```

Comment:
although the id is unique we should make sure a user can only query his own instance using `authz.GetInstance(ctx).InstanceID()`